components/pkg-export-docker/src/build.rs

```diff
@@ -543,11 +544,14 @@ impl BuildRootContext {
         let gid = DEFAULT_USER_AND_GROUP_ID;
 
         let pkg = self.primary_svc()?;
-        let user_name = pkg.svc_user().unwrap_or(Some(String::from("hab"))).unwrap();
+        let user_name = pkg
+            .svc_user()
+            .unwrap_or_default()
+            .unwrap_or_else(|| String::from("hab"));
```

Comment:
[`svc_user`](https://github.com/habitat-sh/core/blob/master/components/core/src/package/install.rs#L579) (and similarly `group` below) return `Result<Option<String>>`, so the first `unwrap_or_default` will generate an `Option::None` and trigger the `or_else` behavior in the case of `Result::Err`.

Reply:
Slick.